src/output/render-manager.cpp

```diff
@@ -849,18 +897,32 @@ class wf::render_manager::impl
 
         for (auto& ds : wf::reverse(repaint.to_render))
         {
+            wlr_surface *sampled_surf = nullptr;
             if (ds->view)
             {
                 repaint.fb.geometry.x = ds->pos.x;
                 repaint.fb.geometry.y = ds->pos.y;
                 ds->view->render_transformed(repaint.fb, ds->damage);
+                sampled_surf = ds->view->get_wlr_surface();
+                for (auto& child : ds->view->enumerate_surfaces({0, 0}))
```

Comment:
braces {} for the `if` and `else` (in general, I have been allowing skipping braces only for short single-line ifs, and I think I'll start required braces even then)